--- a/web/web.go
+++ b/web/web.go
@@ -23,7 +23,6 @@
 	"math"
 	"net"
 	"net/http"
-	"net/http/pprof"
 	"net/url"
 	"os"
 	"path"
@@ -36,7 +35,6 @@
 	"time"
 
 	"github.com/alecthomas/units"
-	"github.com/felixge/fgprof"
 	"github.com/grafana/regexp"
 	"github.com/mwitkow/go-conntrack"
 	remoteapi "github.com/prometheus/client_golang/exp/api/remote"
@@ -114,11 +112,6 @@
 	Stopping
 )
 
-var (
-	fgprofHandler = fgprof.Handler()
-	fgprofMu      sync.Mutex
-)
-
 // withStackTracer logs the stack trace in case the request panics. The function
 // will re-raise the error which will then be handled by the net/http package.
 // It is needed because the go-kit log package doesn't manage properly the
@@ -602,9 +596,6 @@
 		w.Write([]byte("Only POST or PUT requests allowed"))
 	})
 
-	router.Get("/debug/*subpath", serveDebug)
-	router.Post("/debug/*subpath", serveDebug)
-
 	router.Get("/-/healthy", func(w http.ResponseWriter, _ *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		fmt.Fprintf(w, "%s is Healthy.\n", o.AppName)
@@ -623,43 +613,6 @@
 	return h
 }
 
-func serveDebug(w http.ResponseWriter, req *http.Request) {
-	ctx := req.Context()
-	subpath := route.Param(ctx, "subpath")
-
-	if subpath == "/pprof" {
-		http.Redirect(w, req, req.URL.Path+"/", http.StatusMovedPermanently)
-		return
-	}
-
-	if !strings.HasPrefix(subpath, "/pprof/") {
-		http.NotFound(w, req)
-		return
-	}
-	subpath = strings.TrimPrefix(subpath, "/pprof/")
-
-	switch subpath {
-	case "cmdline":
-		pprof.Cmdline(w, req)
-	case "profile":
-		pprof.Profile(w, req)
-	case "symbol":
-		pprof.Symbol(w, req)
-	case "trace":
-		pprof.Trace(w, req)
-	case "fgprof":
-		if !fgprofMu.TryLock() {
-			http.Error(w, "Could not enable fgprof profiling: fgprof profiling already in use", http.StatusInternalServerError)
-			return
-		}
-		defer fgprofMu.Unlock()
-		fgprofHandler.ServeHTTP(w, req)
-	default:
-		req.URL.Path = "/debug/pprof/" + subpath
-		pprof.Index(w, req)
-	}
-}
-
 // SetReady sets the ready status of our web Handler.
 func (h *Handler) SetReady(v ReadyStatus) {
 	if v == Ready {
