diff --git llvm-project.orig/third-party/benchmark/CMakeLists.txt llvm-project/third-party/benchmark/CMakeLists.txt
index d9bcc6a4939b..c9235e0609c0 100644
--- llvm-project.orig/third-party/benchmark/CMakeLists.txt
+++ llvm-project/third-party/benchmark/CMakeLists.txt
@@ -204,6 +204,7 @@ else()
   # Disable warnings regarding deprecated parts of the library while building
   # and testing those parts of the library.
   add_cxx_compiler_flag(-Wno-deprecated-declarations)
+  add_cxx_compiler_flag(-Wno-c2y-extensions)
   if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
     # Intel silently ignores '-Wno-deprecated-declarations',
     # warning no. 1786 must be explicitly disabled.
diff --git llvm-project.orig/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
--- llvm-project.orig/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
+++ llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
@@ -158,6 +158,6 @@
 #include <linux/mroute.h>
 #include <linux/mroute6.h>
-#include <linux/scc.h>
+//#include <linux/scc.h>
 #include <linux/serial.h>
 #include <sys/msg.h>
 #include <sys/ipc.h>
@@ -534,8 +534,8 @@
   unsigned struct_kbsentry_sz = sizeof(struct kbsentry);
   unsigned struct_mtconfiginfo_sz = sizeof(struct mtconfiginfo);
   unsigned struct_nr_parms_struct_sz = sizeof(struct nr_parms_struct);
-  unsigned struct_scc_modem_sz = sizeof(struct scc_modem);
-  unsigned struct_scc_stat_sz = sizeof(struct scc_stat);
+  // unsigned struct_scc_modem_sz = sizeof(struct scc_modem);
+  // unsigned struct_scc_stat_sz = sizeof(struct scc_stat);
   unsigned struct_serial_multiport_struct_sz
       = sizeof(struct serial_multiport_struct);
   unsigned struct_serial_struct_sz = sizeof(struct serial_struct);
