diff --git swift.orig/utils/build-script-impl swift/utils/build-script-impl
index 3140c8bb545..32a0b502ae6 100755
--- swift.orig/utils/build-script-impl
+++ swift/utils/build-script-impl
@@ -2179,7 +2179,7 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DLLDB_ENABLE_CURSES=ON
                     -DLLDB_ENABLE_LIBEDIT=ON
                     -DLLDB_ENABLE_PYTHON=ON
-                    -DLLDB_ENABLE_LZMA=OFF
+                    -DLLDB_ENABLE_LZMA=ON
                     -DLLDB_ENABLE_LUA=OFF
                     -DLLDB_INCLUDE_TESTS:BOOL="${should_configure_tests}"
                     -DLLDB_TEST_LIBCXX_ROOT_DIR:STRING="${libcxx_build_dir}"
