diff -ruN perl-5.42.2.orig/regcomp_study.c perl-5.42.2.new/regcomp_study.c
--- perl-5.42.2.orig/regcomp_study.c	2026-01-18 12:50:04.000000000 -0500
+++ perl-5.42.2.new/regcomp_study.c	2026-07-13 21:12:58.361793182 -0400
@@ -1841,6 +1841,16 @@
                             tail = regnext( tail );
                         }
 
+                        /* The code below currently saves the difference from
+                         * start to finish in a 16-bit field, causing
+                         * GH #23388.  This defeats the design of batching
+                         * tries into chunks that each fit.  khw thinks it is
+                         * too late in the 5.44 cycle to relook at the design,
+                         * so for now anyway, don't make a trie that would
+                         * overflow */
+                        if (tail - startbranch >= U16_MAX) {
+                            continue;
+                        }
 
                         DEBUG_TRIE_COMPILE_r({
                             regprop(RExC_rx, RExC_mysv, tail, NULL, pRExC_state);
diff -ruN perl-5.42.2.orig/t/re/pat_advanced.t perl-5.42.2.new/t/re/pat_advanced.t
--- perl-5.42.2.orig/t/re/pat_advanced.t	2026-01-18 12:50:05.000000000 -0500
+++ perl-5.42.2.new/t/re/pat_advanced.t	2026-07-13 21:12:58.362181020 -0400
@@ -2713,6 +2713,15 @@
         $x =~ s/^[\x{0301}\x{030C}]+//;
     }
 
+    { # GH #23388
+        fresh_perl_is(<<~'PROG', , "", {}, "Avoid trie overflow");
+            my $x = join "|", "aaa".."mzz";
+            my $y = join "|", "naa".."zzz";
+            use re 'Debug';
+            "fnord" =~ m/(?:$x)|(?:$y)/;
+            PROG
+    }
+
 
     # !!! NOTE that tests that aren't at all likely to crash perl should go
     # a ways above, above these last ones.  There's a comment there that, like
