From d14f81d2743aceb8a3651f1b643bda27e442b0e8 Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Fri, 14 Aug 2026 02:08:24 +0200
Subject: [PATCH 16/16] build(Makefile): do not fail on network-legacy missing

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index ee6ff229..333c6304 100644
--- a/Makefile
+++ b/Makefile
@@ -289,6 +289,7 @@ endif
 	mkdir -p $(DESTDIR)${pkgconfigdatadir}
 	install -m 0644 dracut.pc $(DESTDIR)${pkgconfigdatadir}/dracut.pc
 ifneq ($(enable_network_legacy),yes)
+	mkdir -p $(DESTDIR)$(pkglibdir)/modules.d/00network-legacy
 	rm -r $(DESTDIR)$(pkglibdir)/modules.d/[0-9][0-9]network-legacy
 endif
 	if ! [ -n "$(systemdsystemunitdir)" ]; then \
-- 
2.55.0

