--- rustc-beta-src/src/tools/cargo/Cargo.lock.orig	2026-07-02 11:47:42.914176299 -0500
+++ rustc-beta-src/src/tools/cargo/Cargo.lock	2026-07-02 11:48:03.404372146 -0500
@@ -3108,7 +3108,6 @@ checksum = "c9b3acc4b91781bb0b3386669d32
 dependencies = [
  "cc",
  "libc",
- "libssh2-sys",
  "libz-sys",
  "openssl-sys",
  "pkg-config",
@@ -3156,20 +3155,6 @@ dependencies = [
  "pkg-config",
  "vcpkg",
 ]
-
-[[package]]
-name = "libssh2-sys"
-version = "0.3.2"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "c04141a07bb0c0bc461cb657808764de571702a59bc5c726c400ac9a7625e3ab"
-dependencies = [
- "cc",
- "libc",
- "libz-sys",
- "openssl-sys",
- "pkg-config",
- "vcpkg",
-]
 
 [[package]]
 name = "libz-sys"
--- rustc-beta-src/src/tools/cargo/Cargo.toml.orig	2026-07-02 11:47:42.916176318 -0500
+++ rustc-beta-src/src/tools/cargo/Cargo.toml	2026-07-02 11:48:12.603460071 -0500
@@ -51,7 +51,7 @@ filetime = "0.2.27"
 flate2 = { version = "1.1.9", default-features = false, features = ["zlib-rs"] }
 futures = { version = "0.3.32", default-features = false, features = ["std", "executor", "async-await"]}
 futures-timer = "3.0.3"
-git2 = "0.20.4"
+git2 = { version = "0.20.4", default-features = false, features = ["https"] }
 git2-curl = "0.21.0"
 # When updating this, also see if `gix-transport` further down needs updating or some auth-related tests will fail.
 gix = { version = "0.83.0", default-features = false, features = ["sha1", "progress-tree", "parallel", "dirwalk", "status"] }
