--- rustc-beta-src/src/tools/cargo/Cargo.lock.orig	2026-06-02 15:55:01.625762200 -0500
+++ rustc-beta-src/src/tools/cargo/Cargo.lock	2026-06-02 15:56:06.143120013 -0500
@@ -3152,7 +3152,6 @@ version = "0.37.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "b1f111c8c41e7c61a49cd34e44c7619462967221a6443b0ec299e0ac30cfb9b1"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
--- rustc-beta-src/src/tools/cargo/Cargo.toml.orig	2026-06-02 15:54:48.946691882 -0500
+++ rustc-beta-src/src/tools/cargo/Cargo.toml	2026-06-02 15:56:21.480430158 -0500
@@ -88,7 +88,7 @@ proptest = "1.11.0"
 pulldown-cmark = { version = "0.13.3", default-features = false, features = ["html"] }
 rand = "0.10.1"
 regex = "1.12.3"
-rusqlite = { version = "0.39.0", features = ["bundled"] }
+rusqlite = { version = "0.39.0", features = [] }
 rustc-hash = "2.1.2"
 rustc-stable-hash = "0.1.2"
 rustfix = { version = "0.9.7", path = "crates/rustfix" }
