diff -Naur postgresql-18.3/src/backend/utils/misc/postgresql.conf.sample postgresql-18.3_patched/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-18.3/src/backend/utils/misc/postgresql.conf.sample	2026-02-23 22:56:43.000000000 +0100
+++ postgresql-18.3_patched/src/backend/utils/misc/postgresql.conf.sample	2026-05-05 12:34:18.075704453 +0200
@@ -112,7 +112,7 @@
 #ssl_crl_file = ''
 #ssl_crl_dir = ''
 #ssl_key_file = 'server.key'
-#ssl_ciphers = 'HIGH:MEDIUM:+3DES:!aNULL'	# allowed TLSv1.2 ciphers
+ssl_ciphers = 'PROFILE=SYSTEM'	# allowed TLSv1.2 ciphers, default: 'HIGH:MEDIUM:+3DES:!aNULL'
 #ssl_tls13_ciphers = ''	# allowed TLSv1.3 cipher suites, blank for default
 #ssl_prefer_server_ciphers = on
 #ssl_groups = 'X25519:prime256v1'
