--- a/envargs.c	2005-03-04 03:23:38.000000000 +0100
+++ b/envargs.c	2024-11-26 13:17:22.289650230 +0100
@@ -118,7 +118,7 @@
 
             /* remove escape characters */
             while ((argstart = MBSCHR(argstart, '\\')) != (char *)NULL) {
-                strcpy(argstart, argstart + 1);
+                memmove(argstart, argstart + 1, strlen(argstart + 1) + 1);
                 if (*argstart)
                     ++argstart;
             }
