Index: jsoncpp-1.9.7/src/test_lib_json/main.cpp
===================================================================
--- jsoncpp-1.9.7.orig/src/test_lib_json/main.cpp
+++ jsoncpp-1.9.7/src/test_lib_json/main.cpp
@@ -43,6 +43,12 @@ using CharReaderPtr = std::unique_ptr<Js
 static const float kfint32max = float(kint32max);
 static const float kfuint32max = float(kuint32max);
 
+// Construct std::string from u8 literals properly.
+// Needed for changes how C++20 handles unicode string literals.
+std::string operator ""_S(const char8_t* str, std::size_t size) {
+  return { reinterpret_cast< const char* >(str), size };
+}
+
 // //////////////////////////////////////////////////////////////////
 // //////////////////////////////////////////////////////////////////
 // Json Library test cases
@@ -1993,7 +1999,7 @@ JSONTEST_FIXTURE_LOCAL(ValueTest, Static
 
 JSONTEST_FIXTURE_LOCAL(ValueTest, WideString) {
   // https://github.com/open-source-parsers/jsoncpp/issues/756
-  const std::string uni = u8"\u5f0f\uff0c\u8fdb"; // "式，进"
+  const std::string uni = u8"\u5f0f\uff0c\u8fdb"_S; // "式，进"
   std::string styled;
   {
     Json::Value v;
@@ -3110,7 +3116,7 @@ JSONTEST_FIXTURE_LOCAL(ReaderTest, stric
 
 JSONTEST_FIXTURE_LOCAL(ReaderTest, parseChineseWithOneError) {
   checkParse(R"({ "pr)"
-             u8"\u4f50\u85e4" // 佐藤
+             u8"\u4f50\u85e4"_S // 佐藤
              R"(erty" :: "value" })",
              {{18, 19, "Syntax error: value, object or array expected."}},
              "* Line 1, Column 19\n  Syntax error: value, object or array "
@@ -3223,7 +3229,7 @@ JSONTEST_FIXTURE_LOCAL(CharReaderTest, p
     bool ok = reader->parse(doc, doc + std::strlen(doc), &root, &errs);
     JSONTEST_ASSERT(ok);
     JSONTEST_ASSERT(errs.empty());
-    JSONTEST_ASSERT_EQUAL(u8"\u8A2a", root[0].asString()); // "訪"
+    JSONTEST_ASSERT_EQUAL(u8"\u8A2a"_S, root[0].asString()); // "訪"
   }
   {
     char const doc[] = R"([ "\uD801" ])";
@@ -3602,7 +3608,7 @@ struct CharReaderAllowDropNullTest : Jso
   Value emptyArray = Value{Json::arrayValue};
 
   ValueCheck checkEq(const Value& v) {
-    return [=](const Value& root) { JSONTEST_ASSERT_EQUAL(root, v); };
+    return [=, this](const Value& root) { JSONTEST_ASSERT_EQUAL(root, v); };
   }
 
   static ValueCheck objGetAnd(std::string idx, ValueCheck f) {
