From f56815160ef35943c4ddf5fc03acfa7f84e4246f Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 11 May 2011 16:46:57 +0200
Subject: [PATCH] findutils-4.5.7-warnings.patch

---
 xargs/xargs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xargs/xargs.c b/xargs/xargs.c
index da20a58..fbd58ba 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -1396,7 +1396,8 @@ xargs_do_exec (struct buildcmd_control *ctl, void *usercontext, int argc, char *
                  * utility if we run it, for POSIX compliance on the
                  * handling of exit values.
                  */
-                write (fd[1], &saved_errno, sizeof (int));
+                int sink = write (fd[1], &saved_errno, sizeof (int));
+                (void) sink;
               }
 
             close (fd[1]);
-- 
2.55.0

