diff -ur a/m4/posix_spawn_faction_addchdir.m4 b/m4/posix_spawn_faction_addchdir.m4
--- a/m4/posix_spawn_faction_addchdir.m4	2026-01-02 15:46:16.000000000 +0530
+++ b/m4/posix_spawn_faction_addchdir.m4	2026-07-27 23:50:21.890528572 +0530
@@ -1,5 +1,5 @@
 # posix_spawn_faction_addchdir.m4
-# serial 2
+# serial 3
 dnl Copyright (C) 2018-2026 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
@@ -11,10 +11,11 @@
   AC_REQUIRE([gl_SPAWN_H_DEFAULTS])
   AC_REQUIRE([AC_PROG_CC])
   gl_POSIX_SPAWN
-  AC_CHECK_FUNCS_ONCE([posix_spawn_file_actions_addchdir])
+  AC_CHECK_DECLS([posix_spawn_file_actions_addchdir],,,
+    [[#include <spawn.h>]])
   gl_CHECK_FUNCS_ANDROID([posix_spawn_file_actions_addchdir_np],
     [[#include <spawn.h>]])
-  if test $ac_cv_func_posix_spawn_file_actions_addchdir = yes; then
+  if test $ac_cv_have_decl_posix_spawn_file_actions_addchdir = yes; then
     dnl This function is not yet standardized. Therefore override the
     dnl system's implementation always.
     REPLACE_POSIX_SPAWN_FILE_ACTIONS_ADDCHDIR=1
