From f252a9fbe6730c248d9c3756d74ea67ea817eb05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 19 Jun 2026 09:55:22 +0200
Subject: [PATCH] spec: Disable modularity on RHEL >= 11

---
 dnf5.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dnf5.spec b/dnf5.spec
index 8bad60ec..8ae53f3a 100644
--- a/dnf5.spec
+++ b/dnf5.spec
@@ -95,7 +95,13 @@ Provides:       dnf5-command(versionlock)
 
 %bcond_without acl
 %bcond_without comps
+
+%if 0%{?rhel} >= 11
+%bcond_with modulemd
+%else
 %bcond_without modulemd
+%endif
+
 %bcond_without systemd
 
 %bcond_with    html
-- 
2.55.0

