diff -up less-692/lesspipe-2.23/lesspipe.sh.dontdoxml less-692/lesspipe-2.23/lesspipe.sh
--- less-692/lesspipe-2.23/lesspipe.sh.dontdoxml	2026-03-24 22:21:42.400052408 +0100
+++ less-692/lesspipe-2.23/lesspipe.sh	2026-03-24 22:29:34.771427219 +0100
@@ -566,8 +566,7 @@ isfinal () {
 			msg="$x: showing the output of ${cmd[*]}" ;;
 		xml)
 			[[ -z $file2 ]] &&
-			{ { has_cmd xmq && cmd=(isxmq "$1" xml); } ||
-			{ has_htmlprog && cmd=(ishtml "$1"); }; } ;;
+			has_cmd xmr && cmd=(isxmq "$1" xml) ;;
 		html)
 			[[ -z $file2 ]] && has_htmlprog && cmd=(ishtml "$1") ;;
 		dtb|dts)
