diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ca18e72d4..569958cc33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -437,17 +437,6 @@ if (H5_HL_F_SOVERS_EXISTS)
   math (EXPR H5_HL_F_SOVERS_MAJOR ${H5_HL_F_SOVERS_INTERFACE}-${H5_HL_F_SOVERS_RELEASE})
   message (VERBOSE "SOVERSION_HL_F: ${H5_HL_F_SOVERS_MAJOR}.${H5_HL_F_SOVERS_RELEASE}.${H5_HL_F_SOVERS_MINOR}")
 endif ()
-string (REGEX MATCH ".*LT_JAVA_VERS_INTERFACE[ \t]+=[ \t]+([0-9]*).*$" H5_JAVA_SOVERS_EXISTS ${_lt_vers_am_contents})
-if(H5_JAVA_SOVERS_EXISTS)
-  string (REGEX REPLACE ".*LT_JAVA_VERS_INTERFACE[ \t]+=[ \t]+([0-9]*).*$"
-      "\\1" H5_JAVA_SOVERS_INTERFACE ${_lt_vers_am_contents})
-  string (REGEX REPLACE ".*LT_JAVA_VERS_REVISION[ \t]+=[ \t]+([0-9]*).*$"
-      "\\1" H5_JAVA_SOVERS_MINOR ${_lt_vers_am_contents})
-  string (REGEX REPLACE ".*LT_JAVA_VERS_AGE[ \t]+=[ \t]+([0-9]*).*$"
-      "\\1" H5_JAVA_SOVERS_RELEASE ${_lt_vers_am_contents})
-  math (EXPR H5_JAVA_SOVERS_MAJOR ${H5_JAVA_SOVERS_INTERFACE}-${H5_JAVA_SOVERS_RELEASE})
-  message (VERBOSE "SOVERSION_JAVA: ${H5_JAVA_SOVERS_MAJOR}.${H5_JAVA_SOVERS_RELEASE}.${H5_JAVA_SOVERS_MINOR}")
-endif ()
 
 #-----------------------------------------------------------------------------
 # Basic HDF5 stuff here
@@ -508,13 +497,6 @@ else ()
   set (HDF5_HL_F_PACKAGE_SOVERSION "${H5_LIB_SOVERS_MAJOR}.${H5_LIB_SOVERS_RELEASE}.${H5_LIB_SOVERS_MINOR}")
   set (HDF5_HL_F_PACKAGE_SOVERSION_MAJOR "${H5_LIB_SOVERS_MAJOR}")
 endif ()
-if (H5_JAVA_SOVERS_EXISTS)
-  set (HDF5_JAVA_PACKAGE_SOVERSION "${H5_JAVA_SOVERS_MAJOR}.${H5_JAVA_SOVERS_RELEASE}.${H5_JAVA_SOVERS_MINOR}")
-  set (HDF5_JAVA_PACKAGE_SOVERSION_MAJOR "${H5_JAVA_SOVERS_MAJOR}")
-else ()
-  set (HDF5_JAVA_PACKAGE_SOVERSION "${H5_LIB_SOVERS_MAJOR}.${H5_LIB_SOVERS_RELEASE}.${H5_LIB_SOVERS_MINOR}")
-  set (HDF5_JAVA_PACKAGE_SOVERSION_MAJOR "${H5_LIB_SOVERS_MAJOR}")
-endif ()
 set (HDF5_PACKAGE_STRING "${HDF5_PACKAGE_NAME} ${HDF5_PACKAGE_VERSION_STRING}")
 set (HDF5_PACKAGE_TARNAME "${HDF5_PACKAGE}${HDF_PACKAGE_EXT}")
 set (HDF5_PACKAGE_URL "https://www.hdfgroup.org")
diff --git a/config/HDF5Macros.cmake b/config/HDF5Macros.cmake
index 8c30cd0a2a..6f7bffeeee 100644
--- a/config/HDF5Macros.cmake
+++ b/config/HDF5Macros.cmake
@@ -29,7 +29,8 @@
 macro (H5_SET_LIB_OPTIONS libtarget libname libtype libpackage)
   set (LIB_OUT_NAME "${libname}")
   # SOVERSION passed in ARGN when shared
-  if (${libtype} MATCHES "SHARED")
+  # No versioning for Java JNI libraries
+  if (${libtype} MATCHES "SHARED" AND NOT ${libpackage} STREQUAL "JAVA")
     set (PACKAGE_SOVERSION ${HDF5_${libpackage}_PACKAGE_SOVERSION})
     set (PACKAGE_COMPATIBILITY ${H5_${libpackage}_SOVERS_INTERFACE}.0.0)
     set (PACKAGE_CURRENT ${H5_${libpackage}_SOVERS_INTERFACE}.${H5_${libpackage}_SOVERS_MINOR}.0)
diff --git a/config/HDFMacros.cmake b/config/HDFMacros.cmake
index b015e405a3..ecffa41684 100644
--- a/config/HDFMacros.cmake
+++ b/config/HDFMacros.cmake
@@ -435,6 +435,9 @@ macro (HDF_DIR_PATHS package_prefix)
     if (NOT ${package_prefix}_INSTALL_LIB_DIR)
       set (${package_prefix}_INSTALL_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
     endif ()
+    if (NOT ${package_prefix}_INSTALL_JNI_LIB_DIR)
+      set (${package_prefix}_INSTALL_JNI_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
+    endif ()
     if (NOT ${package_prefix}_INSTALL_JAR_DIR)
       set (${package_prefix}_INSTALL_JAR_DIR ${CMAKE_INSTALL_LIBDIR})
     endif ()
diff --git a/java/src-jni/jni/CMakeLists.txt b/java/src-jni/jni/CMakeLists.txt
index 1248bbf4b6..bf9a6b1e0a 100644
--- a/java/src-jni/jni/CMakeLists.txt
+++ b/java/src-jni/jni/CMakeLists.txt
@@ -111,8 +111,8 @@ if (HDF5_EXPORTED_TARGETS)
           ${HDF5_JAVA_JNI_LIB_TARGET}
       EXPORT
           ${HDF5_EXPORTED_TARGETS}_java
-      LIBRARY DESTINATION ${HDF5_INSTALL_LIB_DIR} COMPONENT libraries
-      ARCHIVE DESTINATION ${HDF5_INSTALL_LIB_DIR} COMPONENT libraries
+      LIBRARY DESTINATION ${HDF5_INSTALL_JNI_LIB_DIR} COMPONENT libraries
+      ARCHIVE DESTINATION ${HDF5_INSTALL_JNI_LIB_DIR} COMPONENT libraries
       RUNTIME DESTINATION ${HDF5_INSTALL_BIN_DIR} COMPONENT libraries
       FRAMEWORK DESTINATION ${HDF5_INSTALL_FWRK_DIR} COMPONENT libraries
       INCLUDES DESTINATION include
